snap/snapcraft.yaml

```diff
@@ -0,0 +1,267 @@
+
+  os-prober:
+    command: usr/bin/os-prober
+
+parts:
+  curtin:
+    plugin: nil
+    source: https://git.launchpad.net/curtin
+    source-type: git
+    source-commit: 7c18bf6a24297ed465a341a1f53875b61c878d6b
+    override-pull: |
+      craftctl default
+      PACKAGED_VERSION="$(git describe --long --abbrev=9 --match=[0-9][0-9]*)"
+      sed -e "s,@@PACKAGED_VERSION@@,$PACKAGED_VERSION,g" -i curtin/version.py
+    override-build: &pyinstall |
+      # We install without dependencies because all dependencies come from
+      # archive packages.
+      # XXX: On core22, running `pip3 install --prefix xxx` does not do the
+      # right thing. The package ends up installed in xxx/local and the modules
+      # get installed to dist-packages instead of site-packages.
+      # See https://discuss.python.org/t/18240
+      # As a workaround, we use a fake user install to get the package
+      # installed in the expected place.
+      PYTHONUSERBASE="$CRAFT_PART_INSTALL" pip3 install --user --no-dependencies .
+    build-packages:
+      - python3-pip
+    organize:
+      lib/python*/site-packages/usr/lib/curtin: usr/lib/
+
+  probert:
+    plugin: nil
+    source: https://github.com/canonical/probert.git
+    source-type: git
+    source-commit: ae758355b610e389099abdb2397bb41ad49409ec
+    override-build: *pyinstall
+    build-packages:
+      - build-essential
+      - libnl-3-dev
+      - libnl-genl-3-dev
+      - libnl-route-3-dev
+      - pkg-config
+      - python3-dev
+      - python3-pip
+    build-attributes: [enable-patchelf]
+    stage:
+      - "*"
+      - -bin/python3*
+
+  subiquitydeps:
+    plugin: nil
+    build-attributes: [enable-patchelf]
+    stage-packages:
+      # This list includes the dependencies for curtin and probert as well,
+      # there doesn't seem to be any real benefit to listing them separately.
+      - cloud-init
+      - dctrl-tools
+      - iso-codes
+      - libpython3-stdlib
+      - libpython3.10-minimal
+      - libpython3.10-stdlib
+      - libsystemd0
+      - lsb-release
+      - ntfs-3g
+      - python3-aiohttp
+      - python3-apport
+      - python3-attr
+      - python3-bson
+      - python3-debian
+      - python3-jsonschema
+      - python3-minimal
+      - python3-oauthlib
+      - python3-pkg-resources
+      - python3-pyroute2
+      - python3-pyrsistent
+      - python3-pyudev
+      - python3-requests
+      - python3-requests-unixsocket
+      - python3-systemd
+      - python3-urwid
+      - python3-yaml
+      - python3-yarl
+      - python3.10-minimal
+      - ssh-import-id
+      - ubuntu-advantage-tools
+      # WSL specifics:
+      - language-selector-common
+      - locales
+    prime:
+      - -lib/systemd/system/*
+
+  flutter-git:
+    source: .
+    override-pull: |
+      craftctl default
+      FLUTTER_VERSION=$(sed -n "s/^flutter \([0-9.]\+\).*/\1/p" .tool-versions)
+      git clone -b $FLUTTER_VERSION --depth 1 https://github.com/flutter/flutter.git
```

Comment:
This extracts the flutter version from `.tool-version` and checks out the corresponding git branch of the flutter repo. We've been using this in the firmware-updater for some time now.